website/www/site/content/en/documentation/runtime/environments.md

```diff
@@ -15,56 +15,168 @@ See the License for the specific language governing permissions and
+REPOSITORY                         TAG                 IMAGE ID            CREATED           SIZE
+apache/beam_java8_sdk              latest              ...                 1 min ago         ...
+apache/beam_java11_sdk             latest              ...                 1 min ago         ...
+apache/beam_python3.6_sdk          latest              ...                 1 min ago         ...
+apache/beam_python3.7_sdk          latest              ...                 1 min ago         ...
+apache/beam_python3.8_sdk          latest              ...                 1 min ago         ...
+apache/beam_go_sdk                 latest              ...                 1 min ago         ...
+```
+
+If you did not provide a custom repo/tag as additional parameters (see below), you can retag the image and [push](https://docs.docker.com/engine/reference/commandline/push/) the image using Docker to a remote repository.
+
+```
+export IMAGE_NAME="myrepo/mybeamsdk"
+export TAG="latest"
+
+docker tag apache/beam_python3.6_sdk "${IMAGE_NAME}:${TAG}"
+docker push "${IMAGE_NAME}:${TAG}"
+```
+
+**NOTE**: After pushing a container image, verify the remote image ID and digest matches the local image ID and digest output from `docker_images`
+
+##### Additional Build Parameters
+
+The docker Gradle task defines a default image repository and [tag](https://docs.docker.com/engine/reference/commandline/tag/) is the SDK version defined at [gradle.properties](https://github.com/apache/beam/blob/master/gradle.properties). The default repository is the Docker Hub `apache` namespace, and the default tag is the [SDK version](https://github.com/apache/beam/blob/master/gradle.properties) defined at gradle.properties. With these settings, the
+`docker` command-line tool will implicitly try to push the container to the Docker Hub Apache repository.
+
+You can specify a different repository or tag for built images by providing parameters to the build task. For example:
+
+```
+./gradlew :sdks:python:container:py36:docker -Pdocker-repository-root=example-repo -Pdocker-tag=2019-10-04
+```
 
-To test a customized image locally, run a pipeline with PortableRunner and set the `--environment_config` flag to the image path:
+builds the Python 3.6 container and tags it as `example-repo/beam_python3.6_sdk:2019-10-04`.
+
+From 2.21.0, a `docker-pull-licenses` flag was introduced to add licenses/notices for third party dependencies to the docker images. For example:
+
+```
+./gradlew :sdks:java:container:java8:docker -Pdocker-pull-licenses
+```
+creates a Java 8 SDK image with appropriate licenses in `/opt/apache/beam/third_party_licenses/`.
+
+By default, no licenses/notices are added to the docker images.
+
+
+## Using Container Images in Pipelines
+
+The common method for providing a container image requires using the PortableRunner and setting the `--environment_config` flag to a given image path.
+Other runners, such as Dataflow, support specifying containers with different flags.
```

Comment:
I think in the future there is a plan for Portable runner to support Dataflow endpoint, not sure how far on the roadmap that is.

Reply:
@tvalentyn Can you send me more information on this plan if you have any?

Reply:
+1 I'd like to know about this as well since we shouldn't go to GA with custom containers until the flags are settled.

Reply:
@ananvay @angoenka @robertwb may have input on this plan.

Reply:
For Dataflow, I thought the idea was to use --sdk_harness_container_image or something like this? I'm not sure I have the whole context off the top of my head here, but I'd be happy to chat! :)

Reply:
Regardless of the plans for using the Portability API for submitting jobs, we should standardize on a single flag for specifying the image across runners. It could be argued that environment_config is not the best flag for this; maybe we should take it to the list?

Reply:
> Regardless of the plans for using the Portability API for submitting jobs, we should standardize on a single flag for specifying the image across runners. It could be argued that environment_config is not the best flag for this; maybe we should take it to the list?

+1 I agree that standardizing the flag is a worthwhile improvement, but it's out of scope for this PR.

Reply:
Standardizing the flag should be done before we document and adveritize this feature.

Reply:
Our plan was to keep these as is for preview and standardize for GA. This is also editing existing documentation which used --environment_config. I can send an email to dev re:flags.

Reply:
If we're supporting old SDKs, we have to support the old flag anyway, so we should go ahead and get this in and standardizing going forward.